zipline/pipeline/loaders/quarter_estimates.py

```diff
@@ -0,0 +1,517 @@
+            column.missing_value,
+        )
+
+    def create_overwrite_for_quarter(self,
+                                     next_qtr_start_idx,
+                                     column,
+                                     column_name,
+                                     dates,
+                                     last_per_qtr,
+                                     overwrite,
+                                     quarters_with_estimates_for_sid,
+                                     requested_qtr_data,
+                                     sid,
+                                     sid_idx):
+        # Only add adjustments if the next quarter starts somewhere in
+        # our date index for this sid. Our 'next' quarter can never
+        # start at index 0; a starting index of 0 means that the next
+        # quarter's event date was NaT.
+        if 0 < next_qtr_start_idx < len(dates):
+            # Find the quarter being requested in the quarter we're
+            # crossing into.
+            requested_quarter = requested_qtr_data[
+                SHIFTED_NORMALIZED_QTRS
+            ][sid].iloc[next_qtr_start_idx]
+
+            # If there are estimates for the requested quarter,
+            # overwrite all values going up to the starting index of
+            # that quarter with estimates for that quarter.
+            if requested_quarter in quarters_with_estimates_for_sid:
+                return self.create_overwrite_for_estimate(
+                    column,
+                    column_name,
+                    last_per_qtr,
+                    next_qtr_start_idx,
+                    overwrite,
+                    requested_quarter,
+                    sid,
+                    sid_idx
+                )
+            # There are no estimates for the quarter. Overwrite all
+            # values going up to the starting index of that quarter
+            # with the missing value for this column.
+            else:
+                return self.overwrite_with_null(
+                    column,
+                    last_per_qtr,
+                    next_qtr_start_idx,
+                    overwrite,
+                    sid_idx
+                )
+
+    def overwrite_with_null(self,
+                            column,
+                            last_per_qtr,
+                            next_qtr_start_idx,
+                            overwrite,
+                            sid_idx):
+        return [overwrite(
+                0,
+                next_qtr_start_idx - 1,
+                sid_idx,
+                sid_idx,
+                np.full(
+                    len(
+                        last_per_qtr.index[:next_qtr_start_idx]
+                    ),
+                    column.missing_value,
+                    dtype=column.dtype
+                ))]
+
+    def load_adjusted_array(self, columns, dates, assets, mask):
+        groups = groupby(lambda x: x.dataset.num_quarters, columns)
```

Comment:
could we catch this attribute error and reraise something that tells the user what that attribute is supposed to represent